src/main/scala/com/cognite/sdk/scala/common/OAuth2.scala

```diff
@@ -121,7 +120,11 @@ object OAuth2 {
   // scalastyle:on method.length
 
   object SessionProvider {
-    def apply[F[_]](session: Session, refreshSecondsBeforeTTL: Long = 30)(
+    def apply[F[_]](
+        session: Session,
+        refreshSecondsBeforeTTL: Long = 30,
+        serviceTokenProvider: K8sServiceTokenProvider = K8sServiceToken
```

Comment:
Perhaps make it `getToken: Option[F[String]] = None` and `getToken.orElse(getKubernetesJwt)`, having defined that as `val getKubernetesJwt = Files[F].readAll...` locally.

Reply:
done :)

Reply:
If `Files[F]` turns out to be problematic/difficult, we can "cheat" with Java APIs and `F.delay(Files.readAllBytes(...))`, but don't forget the `F.delay` in that case.